precompile/assetsbridge/min_amount.go

```diff
@@ -207,3 +207,134 @@ func (m *GetMinBridgeOutAmountMethod) Run(
 		precompile.TypesConverter.BigInt.FromSDK(minAmount),
 	}, nil
 }
+
+// SetMinBridgeOutAmountForBitcoinChainMethodName is the name of the setMinBridgeOutAmountForBitcoinChain
+// method. It matches the name of the method in the contract ABI.
+//
+//nolint:gosec
+const SetMinBridgeOutAmountForBitcoinChainMethodName = "setMinBridgeOutAmountForBitcoinChain"
+
+// SetMinBridgeOutAmountForBitcoinChainMethod is the implementation of the setMinBridgeOutAmountForBitcoinChain
+// method.
+type SetMinBridgeOutAmountForBitcoinChainMethod struct {
+	poaKeeper    PoaKeeper
+	bridgeKeeper BridgeKeeper
+}
+
+func newSetMinBridgeOutAmountForBitcoinChainMethod(
+	poaKeeper PoaKeeper,
+	bridgeKeeper BridgeKeeper,
+) *SetMinBridgeOutAmountForBitcoinChainMethod {
+	return &SetMinBridgeOutAmountForBitcoinChainMethod{
+		poaKeeper:    poaKeeper,
+		bridgeKeeper: bridgeKeeper,
+	}
+}
+
+func (m *SetMinBridgeOutAmountForBitcoinChainMethod) MethodName() string {
+	return SetMinBridgeOutAmountForBitcoinChainMethodName
+}
+
+func (m *SetMinBridgeOutAmountForBitcoinChainMethod) MethodType() precompile.MethodType {
+	return precompile.Write
+}
+
+func (m *SetMinBridgeOutAmountForBitcoinChainMethod) RequiredGas(_ []byte) (uint64, bool) {
+	return 0, false
+}
+
+func (m *SetMinBridgeOutAmountForBitcoinChainMethod) Payable() bool {
+	return false
+}
+
+func (m *SetMinBridgeOutAmountForBitcoinChainMethod) Run(
+	context *precompile.RunContext,
+	inputs precompile.MethodInputs,
+) (precompile.MethodOutputs, error) {
+	if err := precompile.ValidateMethodInputsCount(inputs, 1); err != nil {
+		return nil, err
+	}
+
+	minAmount, ok := inputs[0].(*big.Int)
+	if !ok {
+		return nil, fmt.Errorf("invalid minimum amount: %v", inputs[0])
+	}
+
+	if minAmount.Sign() < 0 {
+		return nil, fmt.Errorf("minimum amount must be zero or positive")
+	}
+
+	err := m.poaKeeper.CheckOwner(
+		context.SdkCtx(),
+		precompile.TypesConverter.Address.ToSDK(context.MsgSender()),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	sdkMinAmount, err := precompile.TypesConverter.BigInt.ToSDK(minAmount)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert minimum amount: [%w]", err)
+	}
+
+	err = m.bridgeKeeper.SetMinBridgeOutAmountForBitcoinChain(
+		context.SdkCtx(),
+		sdkMinAmount,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return precompile.MethodOutputs{true}, nil
```

Comment:
not really a big deal, but why return true at all? there's never a case where false can be returned, the method will always revert on error if not mistaken.

Reply:
It matters where precompile functions are called from regular Solidity contracts. If EVM encounters a method without return value, it calls `extcodesize` to see whether the call is against a contract. However, not all of our precompiles may have the bytecode set and if so, the call won't be executed as `extcodesize` returns 0. Because of that, we developed a standard practice to return `bool` from precompile write methods to not bother with this problem at all.

Reply:
Fair enough